src/librustc_driver/lib.rs

```diff
@@ -1179,8 +1184,13 @@ pub fn main() {
             .collect::<Vec<_>>();
         run_compiler(&args, &mut DefaultCallbacks, None, None)
     }).and_then(|result| result);
-    process::exit(match result {
-        Ok(_) => EXIT_SUCCESS,
-        Err(_) => EXIT_FAILURE,
-    });
+    let (show_time, exit_code) = match result {
+        Ok(show_time) => (show_time, EXIT_SUCCESS),
+        Err(_) => (false, EXIT_FAILURE),
+    };
+    // `print_time_passes_entry` increments the depth to 0 before printing.
+    // The extra `\t` is necessary to align this label with the others.
+    set_time_depth(-1);
+    print_time_passes_entry(show_time, "\ttotal", start.elapsed());
```

Comment:
You should change `print_time_passes_entry` to not modify the depth instead.

Reply:
You could also print the crate name with the total, that's useful if you're compiling multiple crates.

Reply:
I have changed `print_time_passes_entry` as you suggest.

I can't work out how to print the crate name. I see `Config::crate_name`, but it gets set to `None` initially and I can't see if/when it is updated.